### DISCUSSION TITLE
Circuit Schematic & Circuit Board Design

### DISCUSSION THREAD

**Discussion by jazzmonger** [jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip#41] [General]
Yes, I'm cheating. I really don't want to debug my first PCB w/ an ESP32 and all the support chips, so I'm using an existing board. Plus, a fully populated board is not cheap and plain circuit boards are dirt cheap!  So I'm using a very small, off the shelf dev board - Wemos ESP32 D1 Mini, (mine are actually clones).  

https://www.amazon.com/ACEIRMC-ESP-WROOM-32-Bluetooth-Development-Compatible/dp/B08PNWB81Z/ref=sr_1_1?s=electronics&sr=1-1
<img width="175" alt="image" src="https://github.com/user-attachments/assets/2ca76422-2954-48e2-b4a5-e853d1fef121">

And I have a dozen level shifters already. 

https://www.amazon.com/gp/product/B07YZTW2SM/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
<img width="150" alt="image" src="https://github.com/user-attachments/assets/0c1acf3e-6fb0-4de2-bb51-31bf9c8423dd">
  
It needs a once over verification, but its all there. I just realized that I tapped Tx & Rx from the TYWE1S chip and not from the header pins.  I think those are buffered/level shifted.  So, I'll need to see if I can eliminate the buffers.  It "should" work on RX as the ESP32 can handle 5v inputs.  I may need to use the 1 leftover level shifter on the TX line though.  I'll test itthat way.

<img width="1222" alt="image" src="https://github.com/user-attachments/assets/247cf1fe-afed-44b6-99b7-8370b1301bf0">

<img width="738" alt="image" src="https://github.com/user-attachments/assets/eddcd67c-5a5a-4ded-8992-32b45f32a45d">

<img width="671" alt="image" src="https://github.com/user-attachments/assets/a51e1fa9-77bb-4787-8b2f-ff8b924f2615">

I was careful to ensure the D1 Mini fits under the housing.  As long as I don't use the plastic part on the header pins it should fit.  I also ordered some low profile headers w/ circular pins to try out on the ESP to get it as close to the board as possible. 
<img width="235" alt="image" src="https://github.com/user-attachments/assets/ab0f5a22-f86a-40c9-8b48-e467dd6d6ba1">

https://www.amazon.com/dp/B0BYWQXYZQ?ref=fed_asin_title

Next its off to JLCPCB to get it made.

**Comment by jazzmonger** [comment]
Ordered!
<img width="834" alt="image" src="https://github.com/user-attachments/assets/351cc692-cbaf-4a21-8975-a542dc8766e3">

**Reply by MadGoatHaz** [reply]
I had your git bookmarked forever ago, decided to use the holiday to finally tackle this... and you're already doin it! I just picked up some d1s... I would love to get involved / help if I can. I have the Ironworks variant of these stoves.

**Reply by jazzmonger** [reply]
I should get circuit boards on Monday - at least that's what Fedex says!  Happy to send you one if it works.  Are you going for the ultimate version or just a reflash of the Tuya board?
for the ultimate you need the ESP32 D1 Mini (not the 8266 version)

**Reply by MadGoatHaz** [reply]
Man I can't tell ya how excited I am to be able to fully tame this quirky beast. I'm gonna go full tilt on it. I just looked to confirm, thought they were d1 they are c6. I have a couple other esp formats but not the d1. --couple d1 on the way

**Reply by jazzmonger** [reply]
Esp32 D1 Mini bc its a full esp32 with 3 full software UARTS which all get used. All the variants C2/C3/S2/S3 etc only have 2. And the circuit board I made is laid out for it. And it's the smallest full Wroom32 dev board I could find. And It all fits in the case. And it's cheap.

But, no other reasons ;)

are you in the USA?

**Reply by MadGoatHaz** [reply]
yes, CO

**Reply by jazzmonger** [reply]
Cool, I'm in Idaho. I'll test the boards when I get them next week and post the results here. 
Have you flashed your Tuya chip yet?

**Reply by MadGoatHaz** [reply]
I haven't, well tried tried is more accurate. I wasn't able to get the tuya local stuff to work . I figured I should just replace the junk if possible. Am I wrong that this is the idea here? Replacement esphome breakout?

**Reply by jazzmonger** [reply]
No, just curious.

**Reply by mannyd209** [reply]
Just ordered the D1 mini. What's your price on a custom board? I'll take one.

**Reply by jazzmonger** [reply]
Let’s wait & make sure they work first!  

I’m just glad others are enjoying this project. 

The boards were $2.68 each. I can ship that along with the 8 pin and 4 pin connectors (male+female) and low profile pins for all the chips for $10. I have several D1s coming as well.

**Comment by jazzmonger** [comment]
Good News, the circuit board worked!  I made a total ROOKIE mistake though and drew a frigging box around the level shifter, and put it on the copper layer instead of the silkscreen layer.  My circuit board trace remover tool came in handy and I just cut the traces, so no big deal.  
![image](https://github.com/user-attachments/assets/6cddc5b0-652b-4ae3-b3d8-0188a8b72c85)

And I had the JST connector backwards on the silkscreen - purely cosmetic.  Or not - depend on how you wire yours!  labels on back of new board now for VCC & GND.  
![image](https://github.com/user-attachments/assets/2898f0f2-c176-4554-aa85-aebda4dce163)

 I had to ditch the socket pins idea.  There's just no room for them, so I used the std header pins to mount the ESP and shifter directly to the board.

Here it is finished and running like a champ.
![image](https://github.com/user-attachments/assets/461e5cc3-0e98-4e85-b548-442471d0e482)

![image](https://github.com/user-attachments/assets/d265c0d0-2e8a-4e4f-b963-d77ff30c3aec)

Here's where I mount the USB connector that goes to the main MCU board for the Auger control.  There's just enough room behind the main power wires JST connector.
![image](https://github.com/user-attachments/assets/484d8c4e-d6f9-4095-a3d1-5de44b1e7f40)

If you want me to send you a kit with the circuit board (new board - see below) along with the JST connectors, use this link.  Choose Flat Rate Shipping and its included for $15.

https://modeltstarters.com/product/pellet-stove-circuit-board/

**Comment by jazzmonger** [comment]
New boards ordered & problems fixed by adding a 2nd level converter for the Auger MCU lines.  I expect them by ~Jan 7-10 or so.
<img width="839" alt="image" src="https://github.com/user-attachments/assets/253e3e1e-dcb7-4227-83a4-c476e854bc98" />

<img width="642" alt="image" src="https://github.com/user-attachments/assets/fe96c438-858c-4304-baab-7f365e59683e" />

**Comment by jazzmonger** [comment]
SUCCESS! Circuit board works great.  I have 5 extras if anyone needs one.

https://modeltstarters.com/product/pellet-stove-circuit-board/
